### ISSUE TITLE
Make grids filtering action work with grids from module extension => switch all existing grids to use filterId feature

### ISSUE THREAD

**Issue by zuk3975** [PrestaShop/PrestaShop#13741]
**Is your feature request related to a problem? Please describe.**
Migrated grids filtering might not work together with grids from modules.
Currently if page consists only one grid, filter request is not parsed to decide which grid is being filtered, so in case any module extends the page with its own grid, the filtering will break. Raised form https://github.com/PrestaShop/PrestaShop/pull/13529#discussion_r281227754

**Describe the solution you'd like**
On every page, regardless if its single or multi grid page, grid id parsing from filtering request should be implemented.

**Comment by khouloudbelguith** [comment]
ping @PrestaShop/prestashop-core-developers

**Comment by jolelievre** [comment]
Hello @zuk3975,
I'm not sure but isn't this use case already managed with this PR https://github.com/PrestaShop/PrestaShop/pull/12928 ?

Or maybe I don't understand correctly, the purpose of this issue is to switch all existing grids using the filterId feature?

**Comment by zuk3975** [comment]
> Hello @zuk3975,
> I'm not sure but isn't this use case already managed with this PR #12928 ?
> 
> Or maybe I don't understand correctly, the purpose of this issue is to switch all existing grids using the filterId feature?

Before #12928 was done, some lists were already migrated without filterId feature, so it might bring some issues in future... So basically, yes, `switch all existing grids to use filterId feature`.

**Comment by zuk3975** [comment]
@jolelievre, in fact it will take a bit more than that. I guess whole filtering could be refactored along with modification of `searchGridAction()` (in PrestaShopBundle\Controller\Admin\CommonController). Then all the `searchAction()` in controllers could be thrown away

**Comment by jolelievre** [comment]
Yeah you're right, but we will be limited in the refacto because we will have to keep some things for backward compatibility
